Core/Src/Channels/RocketChannel.cpp

```diff
@@ -458,37 +435,47 @@ int RocketChannel::getVariable(uint8_t variableId, int32_t &data) const
 	}
 }
 
-void RocketChannel::setRocketState(uint8_t *data, uint8_t &n)
-{
-	RocketStateReqMsg_t *rocketStateRequestMsg;
-	RocketStateResMsg_t *rocketStateResponseMsg;
-	rocketStateRequestMsg = (RocketStateReqMsg_t*) data;
-	externalNextState = static_cast<ROCKET_STATE>(rocketStateRequestMsg->state);
-
-	rocketStateResponseMsg = (RocketStateResMsg_t*) data;
+void RocketChannel::getRocketState(uint8_t *data, uint8_t &n) {
+	RocketStateResMsg_t *rocketStateResponseMsg = (RocketStateResMsg_t*) data;
 	rocketStateResponseMsg->state = state;
-	rocketStateResponseMsg->status = SUCCESS;
+	rocketStateResponseMsg->status = WRITABLE;
```

Comment:
Why don't you differentiate between writeable and write_protected here? Shouldn't you base the response on the ECU type?
```
WRITABLE, //returns on get state when writeable
WRITE_PROTECTED //returns on get state when write protected